Tests/Unit/IoTHubRegistryManagerTests.cs

```diff
@@ -0,0 +1,358 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using global::LoRaTools;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Shared;
+    using Moq;
+    using Xunit;
+
+    public class IoTHubRegistryManagerTests
+    {
+        private readonly MockRepository mockRepository;
+        private readonly Mock<RegistryManager> mockRegistryManager;
+
+        public IoTHubRegistryManagerTests()
+        {
+            this.mockRepository = new MockRepository(MockBehavior.Strict);
+            this.mockRegistryManager = this.mockRepository.Create<RegistryManager>();
+        }
+
+        private IDeviceRegistryManager CreateManager()
+        {
+            return IoTHubRegistryManager.From(this.mockRegistryManager.Object);
+        }
+
+        [Fact]
+        public async Task AddConfigurationAsync()
+        {
+            // Arrange
+            var manager = this.CreateManager();
```

Comment:
All the instances returned from CreateManager are implementing IDisposable, therefore we should change all the occurrencies to:
```suggestion
            using var manager = this.CreateManager();
```
in order to ensure a proper disposal

Reply:
done